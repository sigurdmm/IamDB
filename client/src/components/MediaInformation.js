import React from 'react';
import PropTypes from 'prop-types';
import './InformationSection/index.less';
import MediaType from './InformationSection/MediaType';
import MediaDirector from './InformationSection/MediaDirector';

import InformationSection from './InformationSection';

/**
 * Presents metadata about some media, such as image, type and director
 */
const MediaInformation = ({ details }) => <InformationSection details={details}>
  <MediaType type={details.type}/>
  <MediaDirector director={details.director}/>
</InformationSection>;

MediaInformation.propTypes = {
  details: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default MediaInformation;
