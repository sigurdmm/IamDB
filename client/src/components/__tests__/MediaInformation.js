import { render } from 'enzyme';

import MediaInformation from '../MediaInformation';

const mockedDetails = {
  thumbnails: {
    small: 'image',
  },
  name: 'title',
  rating: 8.8,
  director: 'director',
  type: 'movie',
};

describe('<MediaInformation/>', () => {
  it('should render properly', () => {
    const wrapper = render(<MediaInformation details={mockedDetails}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
