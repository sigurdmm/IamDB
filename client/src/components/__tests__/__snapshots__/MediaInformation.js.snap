// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaInformation/> should render properly 1`] = `
<div
  class="information__container"
>
  <div
    class="mediaimage"
  >
    <img
      class="image__item"
      src="image"
    />
  </div>
  <div
    class="information"
  >
    <div
      class="title"
    >
      <h2>
        title
      </h2>
    </div>
    <h3>
      ⭐️: 8.8
    </h3>
    <h3>
      🎥: movie
    </h3>
    <h3>
      🎬: director
    </h3>
  </div>
</div>
`;
